### PR TITLE
Making SPI Loopback Test more generic

### LINKED ISSUE #4576: no testcase.yaml for tests/drivers/spi/spi_loopback

### PR DESCRIPTION
**Pull request by dwagenk**
When I wanted to add a configuration for stm32f072b_disco board to SPI loopback test in #5405 @ydamigos pointed out, that most of the configuration options where uneeded (because they already derived from boards KConfig.defconfig). This also applies to most of the existing prf_\<board\>.conf files for this test.

I modified the test, to not have many (almost) identical conf files and a bunch of conditional compilation in the source, but use kconfig and a prj_base.conf instead.

I'd appreciate comments, especially regarding these aspects:
### CMakeLists.txt
selection of conf-file(s) is inspired by @superna9999 approach for the i2c_loopback test (#5224), but slightly modified.
Is this or should this be standardized?

### QUARK_SE_C1000_DEVBOARD
These changes schouldn't have any effect when building for this board. Can somebody verify that? <s>I have no knowledge of this board, so I didn't change it.</s> There was a conditional compilation block relying on CONFIG_SOC_QUARK_SE_C1000_SS, but no project configuration for any of the sensor-subsystem boards that define it. I dropped it.

### Frequency when running slow spi test
The default SPI frequency for the slow test was set to 500k for STM32 boards and to 128k for all other boards. STM32 can't go that slow. <s>Would it be OK, to set a default frequency for all boards, that is a little faster (e.g. 256k)?</s> Default now is 500k, boards that had project configuration files for this test and used the 128k speed are configured this way after the Kconfig transition as well.

### Loglevel
Different boards hat different loglevels set, <s>maybe that could be harmonized?</s> Harmonized 

Fixes #4576

### CODE REVIEW

File: tests/drivers/spi/spi_loopback/Kconfig [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+source "$ZEPHYR_BASE/Kconfig.zephyr"
+
+config SPI_LOOPBACK_DRV_NAME
+	string "SPI device name to use for test"
+	default "SPI_1"
+
+config SPI_LOOPBACK_CS_GPIO
+        bool "SPI port CS pin is controlled via a GPIO port during test"
+        depends on GPIO
+        default n
```
Comment (superna9999) [inline tests/drivers/spi/spi_loopback/Kconfig]: These does seem to be aligned the same

**Review by superna9999 — CHANGES_REQUESTED** [review]
Apart these small cosmetic changes, ok with this !

File: tests/drivers/spi/spi_loopback/Kconfig (hunk 1 above)
Comment (DWagenk) [reply tests/drivers/spi/spi_loopback/Kconfig]: fixed

**Review by ydamigos — COMMENTED** [review]
Could you also add a testcase.yaml file?

**Comment by DWagenk** [comment]
@ydamigos I will, but won't get to it this week

**Comment by DWagenk** [comment]
can somebody of the quark devboard users (@SavinayDharmappa that's you, right?) also take a look to verify the changes don't have any (negative) effect on it?

**Comment by nashif** [comment]
please rebase and add a testcase.yaml file per this issue #4576.

**Comment by DWagenk** [comment]
I need assistance on the QUARK_SE_C1000_SS here. (@tbursztyka you introduced this test, right?)
The old test had a section like this:
```c
#elif defined(CONFIG_SOC_QUARK_SE_C1000_SS)

#define SPI_DRV_NAME CONFIG_SPI_0_NAME
#define SPI_SLAVE 0
#define CS_CTRL_GPIO_DRV_NAME CONFIG_GPIO_DW_0_NAME

struct spi_cs_control spi_cs = {
	.gpio_pin = 0,
	.delay = 0
};
```
So i guess the test is meant to support the sensor subsystem, but there's no project configuration to actually compile it for QUARK_SE_C1000_SS_DEVBOARD. Tried builduing it and it failed (latest maaster and this pr with some further modifications).

Can I "drop" support for this SoC, when changing the test? 

Right now in the testcase.yaml I need to blacklist QUARK_SE_C1000_SS_DEVBOARD and ARDUINO_101_SSS for sanitycheck to pass localy

**Comment by tbursztyka** [comment]
@DWagenk Blacklist them for now on. quark_se is going to switch to native driver by default soonish. If the switch works: I'll un-blacklist this test for relevant boards.

**Comment by DWagenk** [comment]
@tbursztyka OK, that's what I did in last commit. Any other comments?
@superna9999 @ydamigos please review again.

**Review by tbursztyka — APPROVED** [review]
Nice changes

**Review by ydamigos — APPROVED** [review]

**Review by superna9999 — APPROVED** [review]
Nice changeset, finally my work goes somewhere !

**Pull request merged**